quarkus-reactive-pg-client not detected when using gradle

### Describe the bug

I cloned this example project: https://github.com/quarkusio/quarkus-quickstarts/tree/main/hibernate-reactive-panache-quickstart, and I replaced maven with gradle.

When I try to build the project:

    gradlew build -x test

the following error occurs:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':quarkusBuild'.
> io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.agroal.deployment.AgroalProcessor#build threw an exception: io.quarkus.runtime.configuration.ConfigurationException: Unable to find a JDBC driver c
orresponding to the database kind 'postgresql' for the default datasource. Either provide a suitable JDBC driver extension, define the driver manually, or disable the JDBC datasource by
adding 'quarkus.datasource.jdbc=false' to your configuration if you don't need it.
        at io.quarkus.agroal.deployment.AgroalProcessor.resolveDriver(AgroalProcessor.java:332)
        at io.quarkus.agroal.deployment.AgroalProcessor.getAggregatedConfigBuildItems(AgroalProcessor.java:278)
        at io.quarkus.agroal.deployment.AgroalProcessor.build(AgroalProcessor.java:83)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:567)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1450)
        at java.base/java.lang.Thread.run(Thread.java:831)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```


**gradle.properties:**

```
quarkusPluginVersion=2.1.0.Final
quarkusPlatformArtifactId=quarkus-bom
quarkusPluginId=io.quarkus
quarkusPlatformGroupId=io.quarkus
quarkusPlatformVersion=2.1.0.Final
org.gradle.warning.mode=all
org.gradle.logging.level=INFO
```

**settings.gradle.kts:**

```
pluginManagement {

    val quarkusPluginId: String by settings
    val quarkusPluginVersion: String by settings

    repositories {
        mavenLocal()
        mavenCentral()
        gradlePluginPortal()
    }

    plugins {
        id(quarkusPluginId) version quarkusPluginVersion
    }
}

rootProject.name = "foo-backend"
```

**build.gradle.kts:**

```
// Project properties can be accessed via delegation
val quarkusPlatformGroupId: String by project
val quarkusPlatformArtifactId: String by project
val quarkusPlatformVersion: String by project

plugins {
    java
    id("io.quarkus")
}

group = "io.fouad.foo"
version = "1.0.0"

repositories {
    mavenCentral()
    mavenLocal()
}

dependencies {
    implementation(enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}"))
    implementation("io.quarkus:quarkus-resteasy-reactive-jackson")
    implementation("io.quarkus:quarkus-resteasy-reactive")
    implementation("io.quarkus:quarkus-hibernate-reactive-panache")
    implementation("io.quarkus:quarkus-reactive-pg-client")
    testImplementation("io.quarkus:quarkus-junit5")
    testImplementation("org.assertj:assertj-core")
    testImplementation("io.rest-assured:rest-assured")
}

tasks {

    java {
        sourceCompatibility = JavaVersion.VERSION_16
        targetCompatibility = JavaVersion.VERSION_16
    }

    compileJava {
        options.encoding = "UTF-8"
        options.compilerArgs = listOf("-parameters", "-Xdoclint:none", "-Xlint:all", "-Xlint:-exports", "-Xlint:-serial", "-Xlint:-try",
                                      "-Xlint:-requires-transitive-automatic", "-Xlint:-requires-automatic", "-Xlint:-processing")
    }

    compileTestJava {
        options.encoding = "UTF-8"
        options.compilerArgs = listOf("-parameters", "-Xdoclint:none", "-Xlint:all", "-Xlint:-exports", "-Xlint:-serial", "-Xlint:-try",
                                      "-Xlint:-requires-transitive-automatic", "-Xlint:-requires-automatic", "-Xlint:-processing")
    }

    jar {
        archiveBaseName.set("foo-backend")
    }

    test {
        useJUnitPlatform()
    }

    quarkusBuild {
        nativeArgs {
            "container-build" to true
        }
    }

}
```


### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

Windows 10

### Output of `java -version`

Java 16

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

Quarkus 2.1.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.1.1

### Additional information

_No response_

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java
@@ -33,7 +33,6 @@ import org.gradle.util.GradleVersion;
 import io.quarkus.gradle.builder.QuarkusModelBuilder;
 import io.quarkus.gradle.dependency.ApplicationDeploymentClasspathBuilder;
 import io.quarkus.gradle.dependency.ConditionalDependenciesEnabler;
-import io.quarkus.gradle.dependency.ExtensionDependency;
 import io.quarkus.gradle.extension.QuarkusPluginExtension;
 import io.quarkus.gradle.extension.SourceSetExtension;
 import io.quarkus.gradle.tasks.QuarkusAddExtension;
@@ -273,15 +272,14 @@ public class QuarkusPlugin implements Plugin<Project> {
         ConditionalDependenciesEnabler conditionalDependenciesEnabler = new ConditionalDependenciesEnabler(project);
         ApplicationDeploymentClasspathBuilder deploymentClasspathBuilder = new ApplicationDeploymentClasspathBuilder(project);
 
-        Set<ExtensionDependency> commonExtensions = conditionalDependenciesEnabler
+        conditionalDependenciesEnabler
                 .declareConditionalDependencies(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME);
-        deploymentClasspathBuilder.createBuildClasspath(commonExtensions, JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME);
-        deploymentClasspathBuilder.addCommonExtension(commonExtensions);
+        deploymentClasspathBuilder.createBuildClasspath(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME, true);
 
         for (String baseConfiguration : CONDITIONAL_DEPENDENCY_LOOKUP) {
-            Set<ExtensionDependency> extensionDependencies = conditionalDependenciesEnabler
+            conditionalDependenciesEnabler
                     .declareConditionalDependencies(baseConfiguration);
-            deploymentClasspathBuilder.createBuildClasspath(extensionDependencies, baseConfiguration);
+            deploymentClasspathBuilder.createBuildClasspath(baseConfiguration, false);
         }
 
         final HashSet<String> visited = new HashSet<>();

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/dependency/ApplicationDeploymentClasspathBuilder.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/dependency/ApplicationDeploymentClasspathBuilder.java
@@ -22,25 +22,25 @@ public class ApplicationDeploymentClasspathBuilder {
         return baseConfigurationName + DEPLOYMENT_CONFIGURATION_SUFFIX;
     }
 
-    public void createBuildClasspath(Set<ExtensionDependency> extensions, String baseConfigurationName) {
+    public void createBuildClasspath(String baseConfigurationName, boolean common) {
         String deploymentConfigurationName = toDeploymentConfigurationName(baseConfigurationName);
         project.getConfigurations().create(deploymentConfigurationName);
 
         DependencyHandler dependencies = project.getDependencies();
-        for (ExtensionDependency extension : extensions) {
-            if (commonExtensions.contains(extension)) {
+        Set<ExtensionDependency> firstLevelExtensions = DependencyUtils.loadQuarkusExtension(project,
+                project.getConfigurations().findByName(baseConfigurationName));
+        for (ExtensionDependency extension : firstLevelExtensions) {
+            if (common) {
+                commonExtensions.add(extension);
+            } else if (commonExtensions.contains(extension)) {
                 continue;
             }
             extension.createDeploymentVariant(dependencies);
-            createDeploymentClasspath(deploymentConfigurationName, extension, dependencies);
+            requireDeploymentDependency(deploymentConfigurationName, extension, dependencies);
         }
     }
 
-    public void addCommonExtension(Set<ExtensionDependency> commonExtensions) {
-        this.commonExtensions.addAll(commonExtensions);
-    }
-
-    private void createDeploymentClasspath(String deploymentConfigurationName, ExtensionDependency extension,
+    private void requireDeploymentDependency(String deploymentConfigurationName, ExtensionDependency extension,
             DependencyHandler dependencies) {
         ExternalDependency dependency = (ExternalDependency) dependencies.add(deploymentConfigurationName,
                 extension.asDependencyNotation());

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/dependency/ConditionalDependenciesEnabler.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/dependency/ConditionalDependenciesEnabler.java
@@ -1,66 +1,46 @@
 package io.quarkus.gradle.dependency;
 
-import java.io.BufferedReader;
-import java.io.File;
-import java.io.IOException;
-import java.nio.file.FileSystem;
-import java.nio.file.Files;
-import java.nio.file.Path;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
-import java.util.Properties;
 import java.util.Set;
 
-import org.gradle.api.GradleException;
 import org.gradle.api.Project;
 import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.artifacts.Dependency;
 import org.gradle.api.artifacts.ModuleVersionIdentifier;
 import org.gradle.api.artifacts.ResolvedArtifact;
 
-import io.quarkus.bootstrap.BootstrapConstants;
-import io.quarkus.bootstrap.model.AppArtifactCoords;
-import io.quarkus.bootstrap.model.AppArtifactKey;
-import io.quarkus.bootstrap.util.BootstrapUtils;
-import io.quarkus.bootstrap.util.ZipUtils;
-
 public class ConditionalDependenciesEnabler {
 
     private final Map<String, ExtensionDependency> featureVariants = new HashMap<>();
+
     private final Project project;
 
     public ConditionalDependenciesEnabler(Project project) {
         this.project = project;
     }
 
-    public Set<ExtensionDependency> declareConditionalDependencies(String baseConfigurationName) {
+    public void declareConditionalDependencies(String baseConfigurationName) {
         featureVariants.clear();
 
         Configuration resolvedConfiguration = DependencyUtils.duplicateConfiguration(project,
                 project.getConfigurations().getByName(baseConfigurationName));
+
         Set<ResolvedArtifact> runtimeArtifacts = resolvedConfiguration.getResolvedConfiguration().getResolvedArtifacts();
         List<ExtensionDependency> extensions = collectExtensionsForResolution(runtimeArtifacts);
         featureVariants.putAll(extractFeatureVariants(extensions));
 
         resolveConditionalDependencies(extensions, resolvedConfiguration, baseConfigurationName);
-
-        Configuration resolvedExtensionsConfiguration = DependencyUtils.duplicateConfiguration(project,
-                project.getConfigurations().getByName(baseConfigurationName));
-        Set<ExtensionDependency> enabledExtension = getEnabledExtension(resolvedExtensionsConfiguration);
-        project.getConfigurations().remove(resolvedExtensionsConfiguration);
-
-        return enabledExtension;
     }
 
     private List<ExtensionDependency> collectExtensionsForResolution(Set<ResolvedArtifact> runtimeArtifacts) {
         List<ExtensionDependency> firstLevelExtensions = new ArrayList<>();
         for (ResolvedArtifact artifact : runtimeArtifacts) {
-            ExtensionDependency extension = getExtensionInfoOrNull(artifact);
+            ExtensionDependency extension = DependencyUtils.getExtensionInfoOrNull(project, artifact);
             if (extension != null) {
                 if (!extension.conditionalDependencies.isEmpty()) {
                     if (extension.needsResolution(runtimeArtifacts)) {
@@ -72,43 +52,19 @@ public class ConditionalDependenciesEnabler {
         return firstLevelExtensions;
     }
 
-    private Set<ExtensionDependency> getEnabledExtension(Configuration classpath) {
-        Set<ExtensionDependency> enabledExtensions = new HashSet<>();
-        for (ResolvedArtifact artifact : classpath.getResolvedConfiguration().getResolvedArtifacts()) {
-            ExtensionDependency extension = getExtensionInfoOrNull(artifact);
-            if (extension != null) {
-                enabledExtensions.add(extension);
-            }
-        }
-        return enabledExtensions;
-    }
-
-    private Map<String, ExtensionDependency> extractFeatureVariants(List<ExtensionDependency> extensions) {
-        Map<String, ExtensionDependency> possibleVariant = new HashMap<>();
-        for (ExtensionDependency extension : extensions) {
-            for (Dependency dependency : extension.conditionalDependencies) {
-                possibleVariant.put(DependencyUtils.asFeatureName(dependency), extension);
-            }
-        }
-        return possibleVariant;
-    }
-
     private void resolveConditionalDependencies(List<ExtensionDependency> conditionalExtensions,
             Configuration existingDependencies, String baseConfigurationName) {
         final Configuration conditionalDeps = createConditionalDependenciesConfiguration(existingDependencies,
                 conditionalExtensions);
         boolean hasChanged = false;
-        Map<String, ExtensionDependency> validConditionalDependencies = new HashMap<>();
         List<ExtensionDependency> newConditionalDependencies = new ArrayList<>();
         newConditionalDependencies.addAll(conditionalExtensions);
         for (ResolvedArtifact artifact : conditionalDeps.getResolvedConfiguration().getResolvedArtifacts()) {
-            ExtensionDependency extensionDependency = getExtensionInfoOrNull(artifact);
+            ExtensionDependency extensionDependency = DependencyUtils.getExtensionInfoOrNull(project, artifact);
             if (extensionDependency != null) {
                 if (DependencyUtils.exist(conditionalDeps.getResolvedConfiguration().getResolvedArtifacts(),
                         extensionDependency.dependencyConditions)) {
                     enableConditionalDependency(extensionDependency.extensionId);
-                    validConditionalDependencies.put(DependencyUtils.asFeatureName(extensionDependency.extensionId),
-                            extensionDependency);
                     if (!extensionDependency.conditionalDependencies.isEmpty()) {
                         featureVariants.putAll(extractFeatureVariants(Collections.singletonList(extensionDependency)));
                     }
@@ -128,6 +84,16 @@ public class ConditionalDependenciesEnabler {
                 resolveConditionalDependencies(newConditionalDependencies, enhancedDependencies, baseConfigurationName);
             }
         }
+    }
+
+    private Map<String, ExtensionDependency> extractFeatureVariants(List<ExtensionDependency> extensions) {
+        Map<String, ExtensionDependency> possibleVariant = new HashMap<>();
+        for (ExtensionDependency extension : extensions) {
+            for (Dependency dependency : extension.conditionalDependencies) {
+                possibleVariant.put(DependencyUtils.asFeatureName(dependency), extension);
+            }
+        }
+        return possibleVariant;
     }
 
     private Configuration createConditionalDependenciesConfiguration(Configuration existingDeps,
@@ -152,56 +118,5 @@ public class ConditionalDependenciesEnabler {
             return;
         }
         extension.importConditionalDependency(project.getDependencies(), dependency);
-    }
-
-    private ExtensionDependency getExtensionInfoOrNull(ResolvedArtifact artifact) {
-        ModuleVersionIdentifier artifactId = artifact.getModuleVersion().getId();
-        File artifactFile = artifact.getFile();
-        if (!artifactFile.exists() || !"jar".equals(artifact.getExtension())) {
-            return null;
-        }
-        if (artifactFile.isDirectory()) {
-            Path descriptorPath = artifactFile.toPath().resolve(BootstrapConstants.DESCRIPTOR_PATH);
-            if (Files.exists(descriptorPath)) {
-                return loadExtensionInfo(descriptorPath, artifactId);
-            }
-        } else {
-            try (FileSystem artifactFs = ZipUtils.newFileSystem(artifactFile.toPath())) {
-                Path descriptorPath = artifactFs.getPath(BootstrapConstants.DESCRIPTOR_PATH);
-                if (Files.exists(descriptorPath)) {
-                    return loadExtensionInfo(descriptorPath, artifactId);
-                }
-            } catch (IOException e) {
-                throw new GradleException("Failed to read " + artifactFile, e);
-            }
-        }
-        return null;
-    }
-
-    private ExtensionDependency loadExtensionInfo(Path descriptorPath, ModuleVersionIdentifier exentionId) {
-        final Properties extensionProperties = new Properties();
-        try (BufferedReader reader = Files.newBufferedReader(descriptorPath)) {
-            extensionProperties.load(reader);
-        } catch (IOException e) {
-            throw new GradleException("Failed to load " + descriptorPath, e);
-        }
-        AppArtifactCoords deploymentModule = AppArtifactCoords
-                .fromString(extensionProperties.getProperty(BootstrapConstants.PROP_DEPLOYMENT_ARTIFACT));
-        final List<Dependency> conditionalDependencies;
-        if (extensionProperties.containsKey(BootstrapConstants.CONDITIONAL_DEPENDENCIES)) {
-            final String[] deps = BootstrapUtils
-                    .splitByWhitespace(extensionProperties.getProperty(BootstrapConstants.CONDITIONAL_DEPENDENCIES));
-            conditionalDependencies = new ArrayList<>(deps.length);
-            for (String conditionalDep : deps) {
-                conditionalDependencies.add(DependencyUtils.create(project.getDependencies(), conditionalDep));
-            }
-        } else {
-            conditionalDependencies = Collections.emptyList();
-        }
-
-        final AppArtifactKey[] constraints = BootstrapUtils
-                .parseDependencyCondition(extensionProperties.getProperty(BootstrapConstants.DEPENDENCY_CONDITION));
-        return new ExtensionDependency(exentionId, deploymentModule, conditionalDependencies,
-                constraints == null ? Collections.emptyList() : Arrays.asList(constraints));
     }
 }

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/dependency/DependencyUtils.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/dependency/DependencyUtils.java
@@ -1,10 +1,20 @@
 package io.quarkus.gradle.dependency;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
+import java.util.Properties;
 import java.util.Set;
 
+import org.gradle.api.GradleException;
 import org.gradle.api.Project;
 import org.gradle.api.UnknownDomainObjectException;
 import org.gradle.api.artifacts.Configuration;
@@ -18,8 +28,11 @@ import org.gradle.api.capabilities.Capability;
 import org.gradle.api.initialization.IncludedBuild;
 import org.gradle.api.plugins.JavaPlugin;
 
+import io.quarkus.bootstrap.BootstrapConstants;
 import io.quarkus.bootstrap.model.AppArtifactCoords;
 import io.quarkus.bootstrap.model.AppArtifactKey;
+import io.quarkus.bootstrap.util.BootstrapUtils;
+import io.quarkus.bootstrap.util.ZipUtils;
 
 public class DependencyUtils {
 
@@ -72,6 +85,21 @@ public class DependencyUtils {
             }
         }
         return false;
+    }
+
+    public static Set<ExtensionDependency> loadQuarkusExtension(Project project, Configuration configuration) {
+        Set<ExtensionDependency> extensions = new HashSet<>();
+        Configuration configurationCopy = duplicateConfiguration(project, configuration);
+
+        Set<ResolvedArtifact> resolvedArtifacts = configurationCopy.getResolvedConfiguration().getResolvedArtifacts();
+        for (ResolvedArtifact artifact : resolvedArtifacts) {
+            ExtensionDependency extension = getExtensionInfoOrNull(project, artifact);
+            if (extension != null) {
+                extensions.add(extension);
+            }
+        }
+
+        return extensions;
     }
 
     public static boolean isTestFixtureDependency(Dependency dependency) {
@@ -137,5 +165,57 @@ public class DependencyUtils {
 
     public static String asFeatureName(Dependency version) {
         return version.getGroup() + ":" + version.getName();
+    }
+
+    public static ExtensionDependency getExtensionInfoOrNull(Project project, ResolvedArtifact artifact) {
+        ModuleVersionIdentifier artifactId = artifact.getModuleVersion().getId();
+        File artifactFile = artifact.getFile();
+        if (!artifactFile.exists() || !"jar".equals(artifact.getExtension())) {
+            return null;
+        }
+        if (artifactFile.isDirectory()) {
+            Path descriptorPath = artifactFile.toPath().resolve(BootstrapConstants.DESCRIPTOR_PATH);
+            if (Files.exists(descriptorPath)) {
+                return loadExtensionInfo(project, descriptorPath, artifactId);
+            }
+        } else {
+            try (FileSystem artifactFs = ZipUtils.newFileSystem(artifactFile.toPath())) {
+                Path descriptorPath = artifactFs.getPath(BootstrapConstants.DESCRIPTOR_PATH);
+                if (Files.exists(descriptorPath)) {
+                    return loadExtensionInfo(project, descriptorPath, artifactId);
+                }
+            } catch (IOException e) {
+                throw new GradleException("Failed to read " + artifactFile, e);
+            }
+        }
+        return null;
+    }
+
+    private static ExtensionDependency loadExtensionInfo(Project project, Path descriptorPath,
+            ModuleVersionIdentifier exentionId) {
+        final Properties extensionProperties = new Properties();
+        try (BufferedReader reader = Files.newBufferedReader(descriptorPath)) {
+            extensionProperties.load(reader);
+        } catch (IOException e) {
+            throw new GradleException("Failed to load " + descriptorPath, e);
+        }
+        AppArtifactCoords deploymentModule = AppArtifactCoords
+                .fromString(extensionProperties.getProperty(BootstrapConstants.PROP_DEPLOYMENT_ARTIFACT));
+        final List<Dependency> conditionalDependencies;
+        if (extensionProperties.containsKey(BootstrapConstants.CONDITIONAL_DEPENDENCIES)) {
+            final String[] deps = BootstrapUtils
+                    .splitByWhitespace(extensionProperties.getProperty(BootstrapConstants.CONDITIONAL_DEPENDENCIES));
+            conditionalDependencies = new ArrayList<>(deps.length);
+            for (String conditionalDep : deps) {
+                conditionalDependencies.add(DependencyUtils.create(project.getDependencies(), conditionalDep));
+            }
+        } else {
+            conditionalDependencies = Collections.emptyList();
+        }
+
+        final AppArtifactKey[] constraints = BootstrapUtils
+                .parseDependencyCondition(extensionProperties.getProperty(BootstrapConstants.DEPENDENCY_CONDITION));
+        return new ExtensionDependency(exentionId, deploymentModule, conditionalDependencies,
+                constraints == null ? Collections.emptyList() : Arrays.asList(constraints));
     }
 }

--- a/integration-tests/gradle/src/main/resources/conditional-dependencies/ext-a/runtime/build.gradle
+++ b/integration-tests/gradle/src/main/resources/conditional-dependencies/ext-a/runtime/build.gradle
@@ -8,6 +8,7 @@ dependencies {
     api("org.acme:simple-dependency:1.0-SNAPSHOT") {
         exclude module: "transitive-dependency"
     }
+    api("io.quarkus:quarkus-hibernate-reactive-panache")
 }
 
 publishing {

--- a/integration-tests/gradle/src/test/java/io/quarkus/gradle/ConditionalDependenciesTest.java
+++ b/integration-tests/gradle/src/test/java/io/quarkus/gradle/ConditionalDependenciesTest.java
@@ -86,6 +86,7 @@ public class ConditionalDependenciesTest extends QuarkusGradleWrapperTestBase {
 
         final Path deploymentLib = buildDir.toPath().resolve("quarkus-app").resolve("lib").resolve("deployment");
         assertThat(deploymentLib.resolve("org.acme.transitive-dependency-1.0-SNAPSHOT.jar")).exists();
+        assertThat(deploymentLib.resolve("io.quarkus.quarkus-agroal-" + getQuarkusVersion() + ".jar")).doesNotExist();
     }
 
     @Test

